### PR TITLE
Add documentation for PKCS#15 profile files

### LINKED ISSUE #1833: Add documentation for PKCS15 *.profile  files
In current release there are plenty of `*.profile` files, but there is no single documentation for it.  Man page is very descriptive: https://github.com/OpenSC/OpenSC/blob/master/doc/files/pkcs15-profile.5.xml

Some **syntatic** information can be extracted from the https://github.com/OpenSC/OpenSC/blob/master/src/pkcs15init/profile.c file, but **semantic** one are almost impossible to figure out.

For someone who is trying to customize the card it is really painful to do it, especially when he is comparing them with PKCS#15 https://www.foo.be/docs/opensst/ref/pkcs/pkcs-15/pkcs-15v11d1.pdf

Also according to existing files themselves, they are formatted in terribly inconsistent way (even in single file there spaces/tabs mixed indents). Another case are magic numbers spreded everywhere.

@viktorTarasov @frankmorgner @okirch  I would be grateful if you will do something with that.

### PR DESCRIPTION
**Pull request by xhanulik**
Fixes #1833

This PR add documentation for PKCS#15 profile files. Any comments or further explanation of some of the options are welcome.

### CODE REVIEW

**Comment by Jakuje** [comment]
Updating HTML files to have "right" paths needs some configure flags `--prefix="/usr"`  as done in the CI check.

File: doc/files/pkcs15-profile.5.xml [hunk 1]
```diff
@@ -34,14 +34,766 @@
...
+				<listitem><para>
+						DF(PKCS#15) is directory containing the PKCS#15 files and directories
+				</para></listitem>
+				<listitem><para>
+						EF(ODF) (Object Directory File) is elementary file containing pointers 
+						to other elementary files (PrKDFs, PuKDFs, SKDFs, CDFs, DODFs, AODFs)
+				</para></listitem>
+				<listitem><para>
+						PrKDF (Private Key Directory File) is elementary file containing
+						pointers to the private keys and additional information information about the private keys
```
Comment (Jakuje) [inline doc/files/pkcs15-profile.5.xml:52]:
```suggestion
						pointers to the private keys and additional information about the private keys
```

File: doc/files/pkcs15-profile.5.xml [hunk 2]
```diff
@@ -34,14 +34,766 @@
...
+				</para></listitem>
+				<listitem><para>
+						PubKDF (Public Key Directory File) is elementary file containing pointers to the public
+						keys and additional information information about the public keys
+				</para></listitem>
+				<listitem><para>
+						CDF (Certificate Directory FIle) is elementary file containing pointers to the
```
Comment (Jakuje) [inline doc/files/pkcs15-profile.5.xml:59]:
```suggestion
						CDF (Certificate Directory File) is elementary file containing pointers to the
```

File: doc/files/pkcs15-profile.5.xml [hunk 3]
```diff
@@ -34,14 +34,766 @@
...
+						certificates and additional information information about the certificates
```
Comment (Jakuje) [inline doc/files/pkcs15-profile.5.xml:60]:
```suggestion
						certificates and additional information about the certificates
```

File: doc/files/pkcs15-profile.5.xml [hunk 4]
```diff
@@ -34,14 +34,766 @@
...
+									<listitem><para>
+											Minimal length of PIN (Default: value <literal>min-pin-length</literal> set in <literal>cardinfo</literal> block).
+									</para></listitem>
+								</varlistentry>
+								<varlistentry>
+									<term>
+										<option>max-length = <replaceable>int</replaceable>;</option>
+									</term>
+									<listitem><para>
+											Minimal length of PIN (Default: value <literal>max-pin-length</literal> set in <literal>cardinfo</literal> block).
```
Comment (Jakuje) [inline doc/files/pkcs15-profile.5.xml:493]:
```suggestion
											Maximal length of PIN (Default: value <literal>max-pin-length</literal> set in <literal>cardinfo</literal> block).
```

File: doc/files/pkcs15-profile.5.xml [hunk 5]
```diff
@@ -34,14 +34,766 @@
...
+														<literal>LINEAR-VARIABLE</literal>
+												</para></listitem>
+												<listitem><para>
+														<literal>LINEAR-VARIABLE-TLV</literal>
+												</para></listitem>
+												<listitem><para>
+														<literal>CYCLIC</literal>
+												</para></listitem>
+												<listitem><para>
+														<literal>CYCLIC-TLVcode</literal>
```
Comment (Jakuje) [inline doc/files/pkcs15-profile.5.xml:687]: I think the `code` was coming from somewhere else:
```suggestion
														<literal>CYCLIC-TLV</literal>
```

**Review by Jakuje — APPROVED** [review]
last two nits, otherwise consider approved.

File: doc/files/pkcs15-profile.5.xml [hunk 6]
```diff
@@ -34,14 +34,766 @@
 			card initialization, such as location of PIN files, key references etc.
 			Profiles currently reside in <filename class="directory">@pkgdatadir@</filename>
```
[the remark this answers is not in the document]
Comment (xhanulik) [reply doc/files/pkcs15-profile.5.xml]: Should be fixed.

**Review by Jakuje — APPROVED** [review]

**Comment by xhanulik** [comment]
Thanks for review.

**Pull request merged**